osu.Game.Rulesets.Osu/Mods/OsuModTransform.cs

```diff
@@ -30,38 +35,92 @@ internal class OsuModTransform : ModWithVisibilityAdjustment
 
         protected override void ApplyNormalVisibilityState(DrawableHitObject hitObject, ArmedState state) => applyTransform(hitObject, state);
 
+        [SettingSource("Movement type", "Alter how circles move into place", 1)]
+        public Bindable<MoveType> Move { get; } = new Bindable<MoveType>();
+
         private void applyTransform(DrawableHitObject drawable, ArmedState state)
         {
-            switch (drawable)
+            switch (Move.Value)
             {
-                case DrawableSliderHead:
-                case DrawableSliderTail:
-                case DrawableSliderTick:
-                case DrawableSliderRepeat:
-                    return;
+                case MoveType.Rotate:
+                    switch (drawable)
```

Comment:
Nested `switch`es - aside, again, putting the idea of having these two modes in one mod into question - are a sign that you need to split methods. Easiest way to do it is have one method for each case in the outer switch.

Reply:
are you suggesting i do all the checks for the drawable parts in the first switch, and make the second one only contain the changes in movements? ill see to that

Reply:
Keep mode checks in `applyTransform()` as one switch, move mode-specific logic with the object type switches to methods.

Reply:
done and thanks for this, much more readable, wow